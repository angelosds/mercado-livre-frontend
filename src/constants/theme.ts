export const ROOT_RELATIVE_SIZE = 16
