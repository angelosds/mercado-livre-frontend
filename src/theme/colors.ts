import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  primary: '#fe6000',
  secondary: '#3483fa',
  gray900: '#333333',
  gray700: '#666666',
  gray500: '#999999',
  gray300: '#eeeeee',
  white: '#ffffff',
};
