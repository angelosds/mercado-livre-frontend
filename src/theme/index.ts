import { DefaultTheme } from 'styled-components';

import { pixelToRem } from '../utils';

export const theme: DefaultTheme = {
  colors: {
    primary500: '#fee600',

    secondary100: '#c6dcf7',
    secondary500: '#3483fa',
    secondary700: '#2968c8',

    gray900: '#333333',
    gray700: '#333333',
    gray500: '#666666',
    gray400: '#999999',
    gray300: '#bbbbbb',
    gray200: '#dddddd',
    gray100: '#eeeeee',

    border200: '#f7f7f7',

    success: '#1ed760',

    white: '#ffffff',
  },
  fontSizes: {
    xs: pixelToRem(14),
    sm: pixelToRem(16),
    md: pixelToRem(24),
    lg: pixelToRem(28),
    xl: pixelToRem(46),
  },
  outline: {
    sm: '3px'
  },
  space: {
    sm: '4px',
    md: '8px',
    lg: '16px',
    xl: '32px',
  },
  radius: {
    sm: '4px'
  },
  breakpoints: {
    mobile: '565px'
  }
};
