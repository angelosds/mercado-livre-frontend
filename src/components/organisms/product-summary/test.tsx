import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect'

import { theme } from '../../../theme';

import ProductSummary, { Props } from '.'

const defaultProps: Props = {
  free_shipping: true,
  picture: 'https://http2.mlstatic.com/D_NQ_NP_721876-MLA46114648081_052021-O.webp',
  title: 'iPhone',
  place: 'SÃ£o Paulo',
  price: 20000,
}

describe('ProductSummary', () => {
  const build = ({ free_shipping, picture, title, place, price }: Props) => {
    const { container } = render(<ThemeProvider theme={theme}><ProductSummary free_shipping={free_shipping} picture={picture} title={title} place={place} price={price} /></ThemeProvider>)

    return { container }
  }

  it('should render the component', () => {
    const { container } = build(defaultProps)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a image', () => {
    build(defaultProps)

    expect(screen.getByRole('img')).toHaveAttribute('src')
  })

  it('should render a product title', () => {
    build(defaultProps)

    expect(screen.getByText(defaultProps.title).firstChild?.textContent).toEqual(defaultProps.title)
  })

  it('should render a product place', () => {
    build(defaultProps)

    expect(screen.getByText(defaultProps.place).firstChild?.textContent).toEqual(defaultProps.place)
  })

  it('should render a product price', () => {
    build(defaultProps)

    expect(screen.getByText(defaultProps.price).firstChild?.textContent).toEqual(defaultProps.price.toString())
  })

  it('should render a truck icon', () => {
    build(defaultProps)

    expect(screen.getByTestId('truck')).toBeVisible()
  })

  it('should render a truck icon', () => {
    build(defaultProps)

    expect(screen.queryByTestId('truck')).toBeTruthy()
  })

  it('should not render a truck icon', () => {
    build({ ...defaultProps, free_shipping: false })

    expect(screen.queryByTestId('truck')).not.toBeTruthy()
  })
})
