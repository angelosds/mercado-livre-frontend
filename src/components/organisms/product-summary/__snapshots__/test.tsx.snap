// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSummary should render the component 1`] = `
<a
  href="/items/MLA902237011"
>
  <div
    class="sc-bdvvtL iqfOnv"
  >
    <div
      class="sc-gsDKAQ enPgso"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 180px; height: 180px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="iPhone"
          data-nimg="fixed"
          decoding="async"
          src="/_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_721876-MLA46114648081_052021-O.webp&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_721876-MLA46114648081_052021-O.webp&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_721876-MLA46114648081_052021-O.webp&w=384&q=75 2x"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
      </span>
    </div>
    <div
      class="sc-dkPtRN"
    >
      <div
        class="sc-hKwDye dEavdP"
      >
        <span
          class="sc-jRQBWg ipLcBZ"
          data-testid="price"
        >
          $ 
          20.000
          <span
            class="sc-iCfMLu lmxByb"
            data-testid="truck"
          >
            <test-file-stub
              height="12"
              width="12"
            />
          </span>
        </span>
        <span
          class="sc-gKclnd bqOqBf"
        >
          SÃ£o Paulo
        </span>
      </div>
      <h1
        class="sc-eCImPb dCpwnM"
      >
        iPhone
      </h1>
    </div>
  </div>
</a>
`;
